ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity-redesigned.component.ts

```diff
@@ -202,4 +210,13 @@ export class ActivityRedesignedComponent extends ActivityComponent implements On
     public isAgree(): boolean {
         return this.model.formType === 'CONSENT' && this.agreeConsent;
     }
+
+    public isReadonly(): boolean {
+        return !this.isAdminEditing && this.model.readonly;
```

Comment:
Thank you for putting this into a method here and avoiding complexity in the template!

Reply:
Clicking this button doesn't change state on the server, it's just helping the user understand that a non-admin user can't see this, and will continue to not see this, right?

Reply:
Yup, it doesn't permanently change the read-only state of activity in server.